src/diffusers/models/unet_2d_condition.py

```diff
@@ -123,20 +124,45 @@ def __init__(
         num_class_embeds: Optional[int] = None,
         upcast_attention: bool = False,
         resnet_time_scale_shift: str = "default",
+        time_embed_dim: int = None,
+        time_embedding_type: str = "positional", # fourier, positional
+        timestep_act_2: bool = False, 
+        class_labels_dim: Optional[int] = None,
+        conv_in_kernel: int = 3,
+        conv_out_kernel: int = 3,
+        norm_out: bool = True,
+        attn1_types: Tuple[bool, None]=(None,'self','self','self'),
+        attn2_types: Tuple[bool, None] = (None, 'cross','cross','cross'),
+        downsample: Optional[Tuple[bool]] = None,
```

Comment:
This should be handled in the downsample and upsample block directly. Let's try to not add a `downsample` parameter for this :-)

Reply:
so the `downsample` and `upsample` parameters actually indicate which down/up blocks contain a downsample/upsample layer ( in our unet currently all down blocks except the final one will have a downsample layer, but in latent-upscaler, all down blocks except the first one have a downsample layer, so it is different) 

I'm not sure how to handle it in the blocks without a parameter passed to it?  but we might be able to find a sensible way to automatically decide it based on other parameters

Reply:
maybe something like this?

```python 
for i, down_block_type in enumerate(down_block_types):
           ...
            down_block = get_down_block(
                down_block_type,
                add_downsample=not is_first_block if down_block_type.startwith('K') else not is_final_block
               ...
            )
            self.down_blocks.append(down_block)
```